be/src/storage/sstable/persistent_index_sstable.cpp

```diff
@@ -17,35 +17,40 @@
 #include <butil/time.h> // NOLINT
 
 #include "fs/fs.h"
-#include "storage/lake/persistent_index_memtable.h"
-#include "storage/lake/tablet_manager.h"
+#include "storage/olap_common.h"
 #include "storage/sstable/table_builder.h"
 #include "util/trace.h"
 
 namespace starrocks::sstable {
 
-Status PersistentIndexSstable::init(RandomAccessFile* rf, const int64_t filesz, Cache* cache) {
+Status PersistentIndexSstable::init(std::unique_ptr<RandomAccessFile> rf, const PersistentIndexSstablePB& sstable_pb,
+                                    Cache* cache) {
     Options options;
     _filter_policy.reset(const_cast<FilterPolicy*>(NewBloomFilterPolicy(10)));
     options.filter_policy = _filter_policy.get();
     options.block_cache = cache;
     Table* table;
-    RETURN_IF_ERROR(Table::Open(options, rf, filesz, &table));
+    RETURN_IF_ERROR(Table::Open(options, rf.get(), sstable_pb.filesz(), &table));
```

Comment:
not sure the scale of the sstable files, looks like every persistentindexsstable will hold a  fd of the file, will it causes too many open fds even the sstable may be not used at all?

Reply:
sstables will be merged into a sstable in major compaction. The size of sstable files should not be too large.

Reply:
leave a comment if possible, let others know the assumption when choosing this design. And later maintainer can properly adjust the design if the assumption can be held anymore.